org.eclipse.xtext.ecore/src/org/eclipse/xtext/ecore/EcoreResourceDescriptionStrategy.java

```diff
@@ -48,8 +47,7 @@ protected boolean createEObjectDescriptions(IQualifiedNameProvider qualifiedName
 		try {
 			QualifiedName qualifiedName = qualifiedNameProvider.getFullyQualifiedName(eObject);
 			if (qualifiedName != null) {
-				Map<String, String> userData = Maps.newHashMapWithExpectedSize(1);
-				userData.put(NS_URI_INDEX_ENTRY, Boolean.toString(isNsURI));
+				Map<String, String> userData = Map.of(NS_URI_INDEX_ENTRY, Boolean.toString(isNsURI));
```

Comment:
I'm reluctant to accept this change, since the map will now throw an NPE when queried with null as a key. Though that's a valid implementation of the Map interface, it doesn't follow the principle of least surprise, which would just be `containsKey(null) == false` and `get(null) == null`

Reply:
In both cases the map is passed to `org.eclipse.xtext.resource.EObjectDescription` and only queried in
https://github.com/eclipse/xtext/blob/2bce8b6c62b1da7c57d81cff6e699eedb19d90a9/org.eclipse.xtext/src/org/eclipse/xtext/resource/EObjectDescription.java#L89-L94


The doc of that method already defines that the name argument must not be null:
https://github.com/eclipse/xtext/blob/2bce8b6c62b1da7c57d81cff6e699eedb19d90a9/org.eclipse.xtext/src/org/eclipse/xtext/resource/IEObjectDescription.java#L49-L54

It was indeed not enforced before, but the new behavior would be 100% compliant with the specification of the method.

Reply:
Understandable. Yet I prefer some simple guidelines in a code base, one of which would be to avoid the overly strict immutable collections. I've seen to many NPE when doing containsKey, indexOf or contains checks on these with a null in code paths that used to work great until they were refactored to the immutable collections.

Reply:
Personally I think it is good that these immutable collections are strict with null elements since one is forced to be more careful in using null, which on the long run and in general is an improvement for a code base, even though it can be annoying in the beginning.

But you prefer not to use it here so I replaced these two usages of Map.of() with `HashMap`.

Reply:
Curious: Why didn't you use Collections.singletonMap instead?
(No need to change this)

Reply:
> Curious: Why didn't you use Collections.singletonMap instead?

I'm not sure anymore. I think I just forgot it during the discussion and initially I probably assumed that `Map.of()` is more elegant while meeting the spec. of the methods.

> (No need to change this)

Even if there is no need, I think that's the better solution and change it accordingly.